spec/system/country_page_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe "Country page", type: :system, driver: :selenium_chrome, js: true do
+  include_context 'countries_setup'
+
+  let(:country_path) { country_path = GeoEntity.last.name.downcase.gsub(/ /, '-') }
+
+  before do
+    allow(GeoEntity).to receive(:permitted_countries).and_return(GeoEntity.all)
+    country
+    many_species
+  
+    visit "/#{country_path}"
+  end
+
+  scenario 'Visiting a country page' do
+    expect(page).to have_content(GeoEntity.last.actual_name)
+  end
+
+  scenario 'Finding the total length of the coastline' do
+    chart_title = find('.habitat-representation__chart-title')
+
+    unrounded_coastline = GeoEntity.last.coastal_stat.total_coast_length
+    coastline_length = ActiveSupport::NumberHelper.number_to_delimited(unrounded_coastline.round)
+
+    expect(chart_title.text).to eq("Percentage of coastline (#{coastline_length} km) covered by habitats")
+  end
```

Comment:
I just want to check this test is testing what it should. Is it an integration test? This test would pass if total_coast_length was calculating the wrong stat (which I'm sure you know). So what it's testing is that the correct methods and formatting are being used to populate the chart_title. Is that all it should be testing? Not sure what convention is here, so just checking :)

Reply:
It's just checking that it actually shows the statistic in the chart title on the page (regardless of any styling used or whether the statistic is wrong). But it will break if the class is changed. Unfortunately that's a problem with these system specs.